TestFileStore fails if files are put on disk too fast

tested on c8bbcb21d1c36a2ba5543f38f45f640edd211858

Test fails. Test succeeds when I add a milisecond delay to write function which makes me think OS timestamp size doesn't have enough resolution for 
```go
	sort.Slice(ids, func(i, j int) bool {
		return ids[i].modTime.Before(ids[j].modTime)
	})
```
to be reliable way to keep it in order



**Describe the bug**
A clear and concise description of what the bug is.
````
--- FAIL: TestFileStore (0.00s)
    store_test.go:73: List returned [2 32300 5890 65535], expected [65535 2 32300 5890]
````

Adding a milisecond delay after write causes this to not show up
It was on XFS filesystem running on NVMe drive.

EDIT: same thing here
```
=== RUN   TestFileQueue
    queue_test.go:80: expected "Queue entry 0 for testing", got "Queue entry 7 for testing"
```

adding delay also helps

**Debug output**

I've redirected test files to `/tmp/asd` for debug:

```
-> ᛯ ls -la /tmp/asd
total 24
drwxr-xr-x  2 xani xani   81 Nov 19 16:59 .
drwxrwxrwt 26 root root 4096 Nov 19 17:00 ..
-rw-------  1 xani xani    8 Nov 19 16:59 foo2.ext
-rw-------  1 xani xani   12 Nov 19 16:59 foo32300.ext
-rw-------  1 xani xani   11 Nov 19 16:59 foo5890.ext
-rw-------  1 xani xani   12 Nov 19 16:59 foo65535.ext
-> ᛯ stat /tmp/asd/foo2.ext    
  File: /tmp/asd/foo2.ext
  Size: 8         	Blocks: 8          IO Block: 4096   regular file
Device: 254,2	Inode: 100675564   Links: 1
Access: (0600/-rw-------)  Uid: ( 1000/    xani)   Gid: ( 1000/    xani)
Access: 2023-11-19 16:59:03.773243442 +0100
Modify: 2023-11-19 16:59:03.773243442 +0100
Change: 2023-11-19 16:59:03.773243442 +0100
 Birth: 2023-11-19 16:59:03.773243442 +0100
[17:00:27] ^ [~/src/os/paho.golang] {master *|u=}
-> ᛯ stat /tmp/asd/foo32300.ext
  File: /tmp/asd/foo32300.ext
  Size: 12        	Blocks: 8          IO Block: 4096   regular file
Device: 254,2	Inode: 100675972   Links: 1
Access: (0600/-rw-------)  Uid: ( 1000/    xani)   Gid: ( 1000/    xani)
Access: 2023-11-19 16:59:03.773243442 +0100
Modify: 2023-11-19 16:59:03.773243442 +0100
Change: 2023-11-19 16:59:03.773243442 +0100
 Birth: 2023-11-19 16:59:03.773243442 +0100
```
Timestamps are "same" on filesystem which causes wrong sorting.

My assumption here is that because there is no Sync() all of the writes are flushed at once hence same timestamp. 
This patch below "fixes" it but I'd wager bug would resurface if ever this code is used on filesystem that have lower timestamp accuracy so I'm hesitant to make a pull request calling it "fixed"

```diff
diff --git a/paho/store/file/store.go b/paho/store/file/store.go
index ed82547..c740a7a 100644
--- a/paho/store/file/store.go
+++ b/paho/store/file/store.go
@@ -86,6 +86,7 @@ func (s *Store) Put(packetID uint16, packetType byte, w io.WriterTo) error {
 		_ = os.Remove(tmpFn)
 		return fmt.Errorf("failed to write packet to temp file: %w", err)
 	}
+	f.Sync()
 	if err = f.Close(); err != nil {
 		_ = os.Remove(tmpFn)
 		return fmt.Errorf("failed to close temp file: %w", err)
```

--- a/paho/store/file/store.go
+++ b/paho/store/file/store.go
@@ -101,6 +101,7 @@ func (s *Store) Put(packetID uint16, packetType byte, w io.WriterTo) error {
 		_ = os.Remove(tmpFn)
 		return fmt.Errorf("failed to write packet to temp file: %w", err)
 	}
+	f.Sync() // want timestamps to be as accurate as possible (close and rename do not imply sync)
 	if err = f.Close(); err != nil {
 		_ = os.Remove(tmpFn)
 		return fmt.Errorf("failed to close temp file: %w", err)

